tests/unit/test.program.js

```diff
@@ -430,6 +435,203 @@ describe('program.main', () => {
+        runOptions: {
+          applyConfigToArgv: fakeApplyConfigToArgv,
+          loadJSConfigFile: fakeLoadJSConfigFile,
+        },
+      })
+      .then(() => {
+        const options = fakeCommands.lint.firstCall.args[0];
+        assert.strictEqual(options.config, fakePath);
+        sinon.assert.calledOnce(fakeLoadJSConfigFile);
+        sinon.assert.calledWith(fakeLoadJSConfigFile,
+                                sinon.match(resolvedFakePath));
+        sinon.assert.calledOnce(fakeApplyConfigToArgv);
+        sinon.assert.calledWith(fakeApplyConfigToArgv, sinon.match({
+          configFileName: resolvedFakePath,
+        }));
+      });
+  });
+
+  it('throws a UsageError when the config file can\'t be loaded', () => {
+    const fakeProcess = createFakeProcess();
+    const fakeCommands = fake(commands, {
+      lint: () => Promise.resolve(),
+    });
+    const fakePath = 'path/to/web-ext-config.js';
+    const fakeLoadJSConfigFile = sinon.spy(() => {
+      throw new UsageError('bad path');
+    });
+    const fakeApplyConfigToArgv = sinon.spy(() => {});
+
+    consoleStream.stopCapturing();
+    consoleStream.flushCapturedLogs();
+    consoleStream.startCapturing();
+
+    return execProgram(
+      ['lint', '--config', fakePath],
+      {
+        commands: fakeCommands,
+        runOptions: {
+          shouldExitProgram: false,
+          systemProcess: fakeProcess,
+          applyConfigToArgv: fakeApplyConfigToArgv,
+          loadJSConfigFile: fakeLoadJSConfigFile,
+        },
+      })
+      .then(makeSureItFails())
+      .catch(onlyInstancesOf(UsageError, (error) => {
+        const {capturedMessages} = consoleStream;
+        consoleStream.stopCapturing();
+
+        assert.match(error.message, /bad path/);
+        assert.ok(
+          capturedMessages.some(
+            (message) => message.match(/bad path/)
+        ));
+        sinon.assert.notCalled(fakeApplyConfigToArgv);
+        sinon.assert.notCalled(fakeProcess.exit);
+      }));
+  });
+
+  it('throws when config file loading raises an unexpected error', () => {
+    const fakeProcess = createFakeProcess();
+    const fakeCommands = fake(commands, {
+      lint: () => Promise.resolve(),
+    });
+    const fakePath = 'path/to/web-ext-config.js';
+    const fakeLoadJSConfigFile = sinon.spy(() => {
+      throw new Error('some error');
+    });
+    const fakeApplyConfigToArgv = sinon.spy(() => {});
+
+    consoleStream.stopCapturing();
+    consoleStream.flushCapturedLogs();
+    consoleStream.startCapturing();
+
+    return execProgram(
+      ['lint', '--config', fakePath],
+      {
+        commands: fakeCommands,
+        runOptions: {
+          shouldExitProgram: false,
+          systemProcess: fakeProcess,
+          applyConfigToArgv: fakeApplyConfigToArgv,
+          loadJSConfigFile: fakeLoadJSConfigFile,
+        },
+      })
+      .then(makeSureItFails())
+      .catch((error) => {
+        if (!(error instanceof UsageError)) {
```

Comment:
@saintsebastian This should be:

```js
const {capturedMessages} = consoleStream;
consoleStream.stopCapturing();

if (!(error instanceof UsageError)) {
  throw error;
}

assert.match(error.message, /some error/);
...
```

to be sure that the test still fails if an UsageError is received here.